### ISSUE TITLE
Docker daemon does not start on camera reboot

### ISSUE THREAD

**Issue by akonanykhin** [AxisCommunications/docker-acap#256]
### Describe the bug

I have installed the docker daemon ACAP v3.0.0 on AXIS P3265-LVE Dome Camera - 22mm. I followed installation instructions for rootless install without TLS and when enabled in the web interface docker works fine, can run containers and everything. Those are the logs (Web Interface -> Apps -> Docker Daemon -> App logs) of a clean run when it is working:

```
----- Contents of SYSTEM_LOG for 'dockerdwrapper' -----


2024-07-18T15:53:28.799+05:00 axis-b8a44f8b757f [ INFO    ] dockerdwrapper[2461]: Starting dockerd without IPC socket and with TCP socket in unsecured mode using /usr/local/packages/dockerdwrapper/localdata/da
2024-07-18T15:53:30.314+05:00 axis-b8a44f8b757f [ INFO    ] dockerdwrapper[2502]: time="2024-07-18T15:53:30.314655554+05:00" level=warning msg="Running in rootless mode. This mode has feature limitations."
2024-07-18T15:53:30.319+05:00 axis-b8a44f8b757f [ INFO    ] dockerdwrapper[2502]: time="2024-07-18T15:53:30.319197275+05:00" level=warning msg="Binding to IP address without --tlsverify is insecure and gives root access on this machine to everyone who has access to your network." host="tcp://0.0.0.0:2375"
2024-07-18T15:53:30.319+05:00 axis-b8a44f8b757f [ INFO    ] dockerdwrapper[2502]: time="2024-07-18T15:53:30.319347756+05:00" level=warning msg="Binding to an IP address, even on localhost, can also give access to scripts run in a browser. Be safe out there!" host="tcp://0.0.0.0:2375"
2024-07-18T15:53:30.319+05:00 axis-b8a44f8b757f [ INFO    ] dockerdwrapper[2502]: time="2024-07-18T15:53:30.319405037+05:00" level=warning msg="[DEPRECATION NOTICE] In future versions this will be a hard failure preventing the daemon from starting! Learn more at: https://docs.docker.com/go/api-security/" host="tcp://0.0.0.0:2375"
2024-07-18T15:53:32.251+05:00 axis-b8a44f8b757f [ INFO    ] dockerdwrapper[2515]: time="2024-07-18T15:53:32.247375649+05:00" level=warning msg="Configuration migrated from version 2, use `containerd config migrate` to avoid migration" t="10.4µs"
2024-07-18T15:53:32.851+05:00 axis-b8a44f8b757f [ INFO    ] dockerdwrapper[2502]: time="2024-07-18T15:53:32.850788021+05:00" level=warning msg="Running modprobe bridge br_netfilter failed with message: modprobe: ERROR: could not insert 'bridge': Operation not permitted\nmodprobe: ERROR: could not insert 'br_netfilter': Operation not permitted\ninsmod /lib/modules/5.15.13-axis9/kernel/net/llc/llc.ko \ninsmod /lib/modules/5.15.13-axis9/kernel/net/llc/llc.ko \n, error: exit status 1"
2024-07-18T15:53:35.806+05:00 axis-b8a44f8b757f [ INFO    ] dockerdwrapper[2502]: time="2024-07-18T15:53:35.806030247+05:00" level=warning msg="Not using native diff for overlay2, this may cause degraded performance for building images: running in a user namespace" storage-driver=overlay2
2024-07-18T15:53:35.807+05:00 axis-b8a44f8b757f [ INFO    ] dockerdwrapper[2502]: time="2024-07-18T15:53:35.806440811+05:00" level=warning msg="[DEPRECATION NOTICE]: API is accessible on http://0.0.0.0:2375 without encryption.\n         Access to the remote API is equivalent to root access on the host. Refer\n         to the 'Docker daemon attack surface' section in the documentation for\n         more information: https://docs.docker.com/go/attack-surface/\nIn future versions this will be a hard failure preventing the daemon from starting! Learn more at: https://docs.docker.com/go/api-security/"
2024-07-18T15:53:35.807+05:00 axis-b8a44f8b757f [ INFO    ] dockerdwrapper[2502]: time="2024-07-18T15:53:35.806540332+05:00" level=warning msg="WARNING: No swap limit support"
2024-07-18T15:53:35.807+05:00 axis-b8a44f8b757f [ INFO    ] dockerdwrapper[2502]: time="2024-07-18T15:53:35.806581892+05:00" level=warning msg="WARNING: No cpuset support"
2024-07-18T15:53:35.807+05:00 axis-b8a44f8b757f [ INFO    ] dockerdwrapper[2502]: time="2024-07-18T15:53:35.806624092+05:00" level=warning msg="WARNING: No io.weight support"
2024-07-18T15:53:35.807+05:00 axis-b8a44f8b757f [ INFO    ] dockerdwrapper[2502]: time="2024-07-18T15:53:35.806666693+05:00" level=warning msg="WARNING: No io.weight (per device) support"
2024-07-18T15:53:35.807+05:00 axis-b8a44f8b757f [ INFO    ] dockerdwrapper[2502]: time="2024-07-18T15:53:35.806705533+05:00" level=warning msg="WARNING: No io.max (rbps) support"
2024-07-18T15:53:35.807+05:00 axis-b8a44f8b757f [ INFO    ] dockerdwrapper[2502]: time="2024-07-18T15:53:35.806742853+05:00" level=warning msg="WARNING: No io.max (wbps) support"
2024-07-18T15:53:35.807+05:00 axis-b8a44f8b757f [ INFO    ] dockerdwrapper[2502]: time="2024-07-18T15:53:35.806782014+05:00" level=warning msg="WARNING: No io.max (riops) support"
2024-07-18T15:53:35.807+05:00 axis-b8a44f8b757f [ INFO    ] dockerdwrapper[2502]: time="2024-07-18T15:53:35.806819534+05:00" level=warning msg="WARNING: No io.max (wiops) support"
2024-07-18T15:53:35.807+05:00 axis-b8a44f8b757f [ INFO    ] dockerdwrapper[2502]: time="2024-07-18T15:53:35.806856494+05:00" level=warning msg="WARNING: bridge-nf-call-iptables is disabled"
2024-07-18T15:53:35.807+05:00 axis-b8a44f8b757f [ INFO    ] dockerdwrapper[2502]: time="2024-07-18T15:53:35.806894975+05:00" level=warning msg="WARNING: bridge-nf-call-ip6tables is disabled"
```

However, when I restart the camera manually or it reboots by itself docker daemon does not start properly, does not respond to commands with a message `Cannot connect to the Docker daemon at tcp://192.168.0.244:2375. Is the docker daemon running?` and shows some new errors in the logs:

```
----- Contents of SYSTEM_LOG for 'dockerdwrapper' -----


2024-07-18T16:00:59.365+05:00 axis-b8a44f8b757f [ INFO    ] dockerdwrapper[1817]: Starting dockerd without IPC socket and with TCP socket in unsecured mode using /usr/local/packages/dockerdwrapper/localdata/da
2024-07-18T16:01:00.881+05:00 axis-b8a44f8b757f [ INFO    ] dockerdwrapper[1888]: time="2024-07-18T16:01:00.881519660+05:00" level=warning msg="Running in rootless mode. This mode has feature limitations."
2024-07-18T16:01:00.889+05:00 axis-b8a44f8b757f [ INFO    ] dockerdwrapper[1888]: time="2024-07-18T16:01:00.889413041+05:00" level=warning msg="Binding to IP address without --tlsverify is insecure and gives root access on this machine to everyone who has access to your network." host="tcp://0.0.0.0:2375"
2024-07-18T16:01:00.889+05:00 axis-b8a44f8b757f [ INFO    ] dockerdwrapper[1888]: time="2024-07-18T16:01:00.889594092+05:00" level=warning msg="Binding to an IP address, even on localhost, can also give access to scripts run in a browser. Be safe out there!" host="tcp://0.0.0.0:2375"
2024-07-18T16:01:00.890+05:00 axis-b8a44f8b757f [ INFO    ] dockerdwrapper[1888]: time="2024-07-18T16:01:00.889675817+05:00" level=warning msg="[DEPRECATION NOTICE] In future versions this will be a hard failure preventing the daemon from starting! Learn more at: https://docs.docker.com/go/api-security/" host="tcp://0.0.0.0:2375"
2024-07-18T16:01:02.764+05:00 axis-b8a44f8b757f [ INFO    ] dockerdwrapper[1908]: time="2024-07-18T16:01:02.760604029+05:00" level=warning msg="Configuration migrated from version 2, use `containerd config migrate` to avoid migration" t="13.241µs"
2024-07-18T16:01:03.435+05:00 axis-b8a44f8b757f [ INFO    ] dockerdwrapper[1888]: time="2024-07-18T16:01:03.435001551+05:00" level=warning msg="Running modprobe bridge br_netfilter failed with message: modprobe: ERROR: could not insert 'bridge': Operation not permitted\nmodprobe: ERROR: could not insert 'br_netfilter': Operation not permitted\ninsmod /lib/modules/5.15.13-axis9/kernel/net/llc/llc.ko \ninsmod /lib/modules/5.15.13-axis9/kernel/net/llc/llc.ko \n, error: exit status 1"
2024-07-18T16:01:04.512+05:00 axis-b8a44f8b757f [ INFO    ] dockerdwrapper[1888]:  (exit status 4)
2024-07-18T16:01:04.512+05:00 axis-b8a44f8b757f [ INFO    ] dockerdwrapper[1888]: failed to start daemon: Error initializing network controller: error obtaining controller instance: failed to register "bridge" driver: failed to create NAT chain DOCKER: iptables failed: iptables -t nat -N DOCKER: Fatal: can't open lock file /run/xtables.lock: Permission denied
2024-07-18T16:01:04.528+05:00 axis-b8a44f8b757f [ INFO    ] dockerdwrapper[1861]: [rootlesskit:child ] error: command [dockerd --config-file /usr/local/packages/dockerdwrapper/localdata/daemon.json --log-level=warn -H tcp://0.0.0.0:2375 --tls=false --data-root /usr/local/packages/dockerdwrapper/localdata/data] exited: exit status 1
2024-07-18T16:01:04.536+05:00 axis-b8a44f8b757f [ INFO    ] dockerdwrapper[1837]: [rootlesskit:parent] error: child exited: exit status 1
```

In the Web interface -> Apps -> Docker Daemon the switch shows as ON. When turned OFF and back ON it starts working again, but the expectation is that daemon should start on its own after reboot.

### To reproduce

1. Download 
Docker_Daemon_3_0_0_aarch64_1320c16_signed.eap from https://github.com/AxisCommunications/docker-acap/releases/tag/3.0.0
2. Web Interface -> Apps -> Add app -> Choose Daemon file
3. In Settings turn off everything but TCP socket
4. Switch it ON in the Web Interface
5. Check with a command `docker -H tcp://192.168.0.244:2375 ps`
6. Working
7. Web Interface -> Maintenance -> Restart
8. Wait for Interface to load + 5 mins
9. Check with a command `docker -H tcp://192.168.0.244:2375 ps`
10. Not working
11. Check Web Interface -> Apps -> Docker Daemon -> App logs
12. Some error messages
13. Turn OFF Docker Daemon and back ON
14. Check with a command `docker -H tcp://192.168.0.244:2375 ps`
15. Working again

### Environment

- Axis device model: P3265-LVE 22mm
- Axis device firmware version: 11.11.73
- Stack trace or logs: [systemlog.txt](https://github.com/user-attachments/files/16282389/systemlog.txt)
- OS and version: Ubuntu 22.04.2 x86_64
- Version: Docker Daemon 3.0.0

Any insights or work-around?

**Comment by madelen-at-work** [comment]
@akonanykhin Thank you for reporting this issue - we will investigate

**Comment by madelen-at-work** [comment]
Duplicate of [#162](https://github.com/AxisCommunications/docker-compose-acap/issues/162)
This issue is fixed in [AXIS OS LTS 2024 11.11](https://help.axis.com/en-us/axis-os-release-notes#lts-2024-11-11) from 11.11.109. Upcoming [AXIS OS 12](https://help.axis.com/en-us/axis-os-release-notes#axis-os-12) will also have the fix.
